### PR TITLE
DOC/MAINT: refguide / doctest drudgery

### PR DESCRIPTION
**Pull request by pv**
Some drudgery as a continuation of gh-4892
- Adjustments to `__all__`  (please review! MAINT commits at the end)
- Additions to refguide (ensure everything in `__all__` is there)
- Fix up doctests so that all of them run
- Turn on refguide+doctest checking on Travis-CI

### CODE REVIEW

**Comment by rgommers** [comment]
Whoa, that looks good!

**Comment by pv** [comment]
Some of the examples don't run on Python 3 --- but I'd leave fixing that for later, not this PR

File: scipy/constants/__init__.py [hunk 1]
```diff
@@ -11,35 +11,48 @@
...
 
-============  =================================================================
-``pi``        Pi
-``golden``    Golden ratio
-============  =================================================================
+===============  =================================================================
+``pi``           Pi
+``golden``       Golden ratio
+``golden_ratio`` Golden ratio
+===============  =================================================================
```
Comment (charris) [inline scipy/constants/__init__.py]: Looks like one `=`  short on the left.

File: scipy/constants/__init__.py [hunk 2]
```diff
@@ -213,6 +240,7 @@
 ``kmh``              kilometers per hour in meters per second
 ``mph``              miles per hour in meters per second
 ``mach``             one Mach (approx., at 15 C, 1 atm) in meters per second
+``speed_of_sound``   one Mach (approx., at 15 C, 1 atm) in meters per second
 ``knot``             one knot in meters per second
 =================    ==========================================================
```
Comment (charris) [inline scipy/constants/__init__.py]: Too short `=` bit?

**Comment by pv** [comment]
Added automatic RST syntax checking now that we're at it.

**Comment by rgommers** [comment]
Needs a rebase - guess that'll happen every other day with this PR due to its size. So I'll try to merge it asap after the rebase.

**Comment by argriffing** [comment]
> please review!

Thanks for doing this work! I've paged through it and couldn't see a reason not to merge soon after rebasing as @rgommers has suggested.

**Comment by pv** [comment]
Ok, done.

**Comment by rgommers** [comment]
Allright, in it goes. Thanks @pv!

**Pull request merged**